DopplerFiles/Controllers/ManagerController.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using StorageProviders;
+using System.Threading.Tasks;
+
+namespace DopplerFiles.Controllers
+{
+    [ApiController]
+    [Authorize]
+    public class ManagerController : ControllerBase
+    {
+        private readonly IStorageProvider _storageProvider;
+        private readonly IFileValidator _fileValidator;
+        private readonly IOptions<ServiceSettings> _serviceSettings;
+
+        public ManagerController(IStorageProvider storageProvider, IFileValidator fileValidator, IOptions<ServiceSettings> serviceSettings)
+        {
+            _storageProvider = storageProvider;
+            _fileValidator = fileValidator;
+            _serviceSettings = serviceSettings;
+        }
+
+        [HttpPost]
+        [Route("[controller]/{idUser?}")]
+        public async Task<ActionResult> UploadFile([FromBody] UploadFileRequest request, string idUser = null)
+        {
+            var error = _fileValidator.IsValid(request.PathFile, request.Content);
+            if (error != StorageProviderError.None)
+            {
+                return BadRequest(error);
+            }
+
+            var result = await _storageProvider.UploadFile(request.PathFile, idUser ?? _serviceSettings.Value.SuperUserFolder, request.Content, request.Override);
+
+            if (result.StorageProviderError != StorageProviderError.None)
+            {
+                return BadRequest(result.StorageProviderError.ToString());
+            }
+
+            return Ok(result.PathFile);
+        }
+
+        [HttpGet]
+        [AllowAnonymous]
+        [Route("[controller]/{keyFile}")]
+        public ActionResult GetDownloadUrl(string keyFile)
```

Comment:
Where the keyfile comes from? I mean, who generates it? 

I am not seeing the userid section in the URL, how to restrict a user to get a file from another user?

Reply:
KeyFile is provided by the upload endpoint and it will save for example in the form model to be used at render form time to get the download URL. A user is not needed because every time a form is a renderer a new temporally download access URL is generated : 
**https://doppler-files-int.s3.us-east-2.amazonaws.com/MS%20Logo.png?X-Amz-Expires=43200&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAUGNR6ZIDNJ75K6PA/20210312/us-east-2/s3/aws4_request&X-Amz-Date=20210312T132828Z&X-Amz-SignedHeaders=host&X-Amz-Signature=_d9b884b09200ecd08cfe54bb9a8bf0bc140a57bdeb193020c9f9db3a20b8a820_**.  where is different every time. Yes, the URL can be used outside of the form by any user because is public but for a  shorter period of time, and never the URL generated is the same.  Keep in mind that the published form is for public access as well.

Reply:
In that case, I am not understanding the API.

I think that you should prepare a little demo including @dnoya, me, and other interested people.